src/renderer/components/+preferences/app-preferences/get-app-preferences.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { computed, IComputedValue } from "mobx";
+import type { LensRendererExtension } from "../../../../extensions/lens-renderer-extension";
+import type { AppPreferenceRegistration } from "./app-preference-registration";
+
+interface Dependencies {
+  extensions: IComputedValue<LensRendererExtension[]>;
+}
+
+interface RegisteredAppPreference extends AppPreferenceRegistration {
+  id: string;
+}
+
+function  getRegisteredItem(item: AppPreferenceRegistration): RegisteredAppPreference {
+  return {
+    id: item.id || item.title.toLowerCase().replace(/[^0-9a-zA-Z]+/g, "-"),
+    ...item,
+  };
+}
+
+
+export const getAppPreferences = ({ extensions }: Dependencies) => {
+  return computed(() => {
+    return [
+      ...extensions.get()
+        .flatMap((extension) => extension.appPreferences)
+        .map(pref => getRegisteredItem(pref)),
+    ];},
+  );
```

Comment:
nit, there is never a need to spread a `.map()` call.

```suggestion
  return computed(() => (
    extensions.get()
      .flatMap((extension) => extension.appPreferences)
      .map(getRegisteredItem)
  ));
```

Reply:
good catch. I was duplicating what was done in https://github.com/lensapp/lens/pull/4615, but there they have another element in the `[ ]`